components/__tests__/SingleReport.test.js

```diff
@@ -0,0 +1,78 @@
+import { mount, flushPromises } from "@vue/test-utils";
+import SingleReport from "../SingleReport.vue";
+import { expect } from "vitest";
+import { SINGLE_REPORT_MOCK } from "../../utils/mocks";
+import { VueQueryPlugin } from "@tanstack/vue-query";
+import {
+  findAxoSwitchToggledProp,
+  findSwitch,
+} from "../../utils/__tests__/test-utils";
+
+let wrapper;
+
+const createComponent = ({ props = {}, shallow = false } = {}) => {
+  wrapper = mount(SingleReport, {
+    props,
+    shallow,
+    global: {
+      plugins: [VueQueryPlugin],
+    },
+  });
+};
+
+const currentPkg = SINGLE_REPORT_MOCK.suggestions[0];
+
+describe("SingleReport component", () => {
+  beforeEach(async () => {
+    createComponent({
+      props: {
+        report: currentPkg,
+        criteria: SINGLE_REPORT_MOCK.criteria,
+      },
+    });
+    await flushPromises();
+  });
+
+  afterEach(() => {
+    wrapper.unmount();
+    wrapper = null;
+  });
+
+  it("mounts and shows package correctly", async () => {
+    expect(wrapper.find("h1").text()).toContain(currentPkg.name);
+    expect(wrapper.find("small").text()).toContain(
+      `changed from v${currentPkg.suggested_diff.from} to v${currentPkg.suggested_diff.to}`
+    );
+  });
+
+  it("shows correct switch state and toggles it", async () => {
+    expect(findAxoSwitchToggledProp(wrapper)).toBe(true);
+    await findSwitch(wrapper).trigger("click");
+    expect(findAxoSwitchToggledProp(wrapper)).toBe(false);
+  });
```

Comment:
ref: https://github.com/axodotdev/harbor/pull/43/files#r1042133493

Added some switch tests, I have no idea why it wasn't working before

Reply:
What do you mean?

Reply:
I meant you could test that for something where the test was default toggled to `on`, you could test that from this component, since this is where that data comes from, right? And that this is testing just that the switch works and probably isn't needed.

Reply:
I'm gonna leave it, I think it's alaways an asset to have this aswell since it's a small test and it helps make sure we cover everything